### ISSUE TITLE
[iOS] Crashed: com.apple.main-thread - [FlutterEngine iosPlatformView]

### ISSUE THREAD

**Issue by siddew201** [flutter/flutter#112545]
We are getting crash in production and it is not reproducible. Any help is appreciable.

Flutter 3.0.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision ee4e09cce0 (5 months ago) • 2022-05-09 16:45:18 -0700
Engine • revision d1b9a6938a
Tools • Dart 2.17.0 • DevTools 2.12.2

CrashLog:
```
Crashed: com.apple.main-thread
0  Flutter                        0x10c90 -[FlutterEngine iosPlatformView] + 53 (weak_ptr.h:53)
1  Flutter                        0x112e4 -[FlutterEngine attachView] + 381 (FlutterEngine.mm:381)
2  Flutter                        0x2d3e4 -[FlutterViewController viewDidLoad] + 671 (FlutterViewController.mm:671)
3  TCPApp                         0x983df0 viewDidLoad + 30 (NeuFlutterViewController.swift:30)
4  TCPApp                         0x983ebc viewDidLoad + 4346740412 (<compiler-generated>:4346740412)
5  UIKitCore                      0x1abffc -[UIViewController _sendViewDidLoadWithAppearanceProxyObjectTaggingEnabled] + 108
6  UIKitCore                      0x1aeb4c -[UIViewController loadViewIfRequired] + 1048
7  UIKitCore                      0x181bac -[UIViewController view] + 32
8  Flutter                        0x3893c flutter::AccessibilityBridge::~AccessibilityBridge() + 66 (accessibility_bridge.mm:66)
9  Flutter                        0x46050 std::__1::unique_ptr<flutter::AccessibilityBridge, std::__1::default_delete<flutter::AccessibilityBridge>>::reset(flutter::AccessibilityBridge*) + 2267 (memory:2267)
10 Flutter                        0x46130 flutter::PlatformViewIOS::~PlatformViewIOS() + 66 (platform_view_ios.mm:66)
11 Flutter                        0x44d14 flutter::PlatformViewIOS::~PlatformViewIOS() + 66 (platform_view_ios.mm:66)
12 Flutter                        0x3e92c0 std::__1::__function::__func<fml::internal::CopyableLambda<flutter::Shell::~Shell()::$_6>, std::__1::allocator<fml::internal::CopyableLambda<flutter::Shell::~Shell()::$_6>>, void ()>::operator()() + 502 (shell.cc:502)
13 Flutter                        0x3e8bf0 flutter::Shell::~Shell() + 1838 (functional:1838)
14 Flutter                        0x157bc std::__1::unique_ptr<flutter::Shell, std::__1::default_delete<flutter::Shell>>::reset(flutter::Shell*) + 2267 (memory:2267)
15 Flutter                        0x114f0 -[FlutterEngine destroyContext] + 4076 (type_traits:4076)
16 CoreFoundation                 0x2a834 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 28
17 CoreFoundation                 0xc6fd4 ___CFXRegistrationPost_block_invoke + 52
18 CoreFoundation                 0x9a1d0 _CFXRegistrationPost + 456
19 CoreFoundation                 0x408ac _CFXNotificationPost + 728
20 Foundation                     0x1b734 -[NSNotificationCenter postNotificationName:object:userInfo:] + 96
21 UIKitCore                      0xf5bde4 -[UIApplication _terminateWithStatus:] + 300
22 UIKitCore                      0x6369d4 -[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:] + 128
23 UIKitCore                      0x82c834 -[_UISceneLifecycleMultiplexer forceExitWithTransitionContext:scene:] + 224
24 UIKitCore                      0xf58128 -[UIApplication workspaceShouldExit:withTransitionContext:] + 224
25 FrontBoardServices             0x4efec __63-[FBSWorkspaceScenesClient willTerminateWithTransitionContext:]_block_invoke_2 + 80
26 FrontBoardServices             0xa308 -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 240
27 FrontBoardServices             0x4ef84 __63-[FBSWorkspaceScenesClient willTerminateWithTransitionContext:]_block_invoke + 132
28 libdispatch.dylib              0x3a30 _dispatch_client_callout + 20
29 libdispatch.dylib              0x74e0 _dispatch_block_invoke_direct + 264
30 FrontBoardServices             0xbc70 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 48
31 FrontBoardServices             0xb040 -[FBSSerialQueue _targetQueue_performNextIfPossible] + 220
32 FrontBoardServices             0xf700 -[FBSSerialQueue _performNextFromRunLoopSource] + 28
33 CoreFoundation                 0xbb414 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
34 CoreFoundation                 0xcc1a0 __CFRunLoopDoSource0 + 208
35 CoreFoundation                 0x56f8 __CFRunLoopDoSources0 + 368
36 CoreFoundation                 0xb05c __CFRunLoopRun + 828
37 CoreFoundation                 0x1ebc8 CFRunLoopRunSpecific + 600
38 GraphicsServices               0x1374 GSEventRunModal + 164
39 UIKitCore                      0x514b58 -[UIApplication _run] + 1100
40 UIKitCore                      0x296090 UIApplicationMain + 364
41 TCPApp                         0xbb0610 main + 39 (AppDelegate.swift:39)
42 ???                            0x107141da4 (Missing)
```

**Comment by huycozy** [comment]
@flutter-symbolizer-bot https://github.com/flutter/flutter/issues/112545#issue-1389272045 force flutter#v3.0.0 ios release arm64

**Comment by huycozy** [comment]
@flutter-symbolizer-bot this force ios flutter#v3.0.0 arm64 release

```
Crashed: com.apple.main-thread
0  Flutter                        0x10c90 -[FlutterEngine iosPlatformView] + 53 (weak_ptr.h:53)
1  Flutter                        0x112e4 -[FlutterEngine attachView] + 381 (FlutterEngine.mm:381)
2  Flutter                        0x2d3e4 -[FlutterViewController viewDidLoad] + 671 (FlutterViewController.mm:671)
3  TCPApp                         0x983df0 viewDidLoad + 30 (NeuFlutterViewController.swift:30)
4  TCPApp                         0x983ebc viewDidLoad + 4346740412 (<compiler-generated>:4346740412)
5  UIKitCore                      0x1abffc -[UIViewController _sendViewDidLoadWithAppearanceProxyObjectTaggingEnabled] + 108
6  UIKitCore                      0x1aeb4c -[UIViewController loadViewIfRequired] + 1048
7  UIKitCore                      0x181bac -[UIViewController view] + 32
8  Flutter                        0x3893c flutter::AccessibilityBridge::~AccessibilityBridge() + 66 (accessibility_bridge.mm:66)
9  Flutter                        0x46050 std::__1::unique_ptr<flutter::AccessibilityBridge, std::__1::default_delete<flutter::AccessibilityBridge>>::reset(flutter::AccessibilityBridge*) + 2267 (memory:2267)
10 Flutter                        0x46130 flutter::PlatformViewIOS::~PlatformViewIOS() + 66 (platform_view_ios.mm:66)
11 Flutter                        0x44d14 flutter::PlatformViewIOS::~PlatformViewIOS() + 66 (platform_view_ios.mm:66)
12 Flutter                        0x3e92c0 std::__1::__function::__func<fml::internal::CopyableLambda<flutter::Shell::~Shell()::$_6>, std::__1::allocator<fml::internal::CopyableLambda<flutter::Shell::~Shell()::$_6>>, void ()>::operator()() + 502 (shell.cc:502)
13 Flutter                        0x3e8bf0 flutter::Shell::~Shell() + 1838 (functional:1838)
14 Flutter                        0x157bc std::__1::unique_ptr<flutter::Shell, std::__1::default_delete<flutter::Shell>>::reset(flutter::Shell*) + 2267 (memory:2267)
15 Flutter                        0x114f0 -[FlutterEngine destroyContext] + 4076 (type_traits:4076)
16 CoreFoundation                 0x2a834 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 28
17 CoreFoundation                 0xc6fd4 ___CFXRegistrationPost_block_invoke + 52
18 CoreFoundation                 0x9a1d0 _CFXRegistrationPost + 456
19 CoreFoundation                 0x408ac _CFXNotificationPost + 728
20 Foundation                     0x1b734 -[NSNotificationCenter postNotificationName:object:userInfo:] + 96
21 UIKitCore                      0xf5bde4 -[UIApplication _terminateWithStatus:] + 300
22 UIKitCore                      0x6369d4 -[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:] + 128
23 UIKitCore                      0x82c834 -[_UISceneLifecycleMultiplexer forceExitWithTransitionContext:scene:] + 224
24 UIKitCore                      0xf58128 -[UIApplication workspaceShouldExit:withTransitionContext:] + 224
25 FrontBoardServices             0x4efec __63-[FBSWorkspaceScenesClient willTerminateWithTransitionContext:]_block_invoke_2 + 80
26 FrontBoardServices             0xa308 -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 240
27 FrontBoardServices             0x4ef84 __63-[FBSWorkspaceScenesClient willTerminateWithTransitionContext:]_block_invoke + 132
28 libdispatch.dylib              0x3a30 _dispatch_client_callout + 20
29 libdispatch.dylib              0x74e0 _dispatch_block_invoke_direct + 264
30 FrontBoardServices             0xbc70 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 48
31 FrontBoardServices             0xb040 -[FBSSerialQueue _targetQueue_performNextIfPossible] + 220
32 FrontBoardServices             0xf700 -[FBSSerialQueue _performNextFromRunLoopSource] + 28
33 CoreFoundation                 0xbb414 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
34 CoreFoundation                 0xcc1a0 __CFRunLoopDoSource0 + 208
35 CoreFoundation                 0x56f8 __CFRunLoopDoSources0 + 368
36 CoreFoundation                 0xb05c __CFRunLoopRun + 828
37 CoreFoundation                 0x1ebc8 CFRunLoopRunSpecific + 600
38 GraphicsServices               0x1374 GSEventRunModal + 164
39 UIKitCore                      0x514b58 -[UIApplication _run] + 1100
40 UIKitCore                      0x296090 UIApplicationMain + 364
41 TCPApp                         0xbb0610 main + 39 (AppDelegate.swift:39)
42 ???                            0x107141da4 (Missing)
```

**Comment by mraleph** [comment]
@huycozy the crash is already symbolized no need to resymbolise it

**Comment by huycozy** [comment]
@siddew201 
Could you please provide `pubspec.yaml` file content and a list of affected device information? 
Also, please provide the output of `flutter doctor -v` as well. Thank you!

**Comment by siddew201** [comment]
Hi @huycozy  Thanks for response . Please find pubspec.yml file content  and affected device/OS detail below

**Pubspec.yml content**

version: 1.0.0+1

environment:   sdk: ">=2.16.2 <3.0.0"

dependencies:   flutter:     sdk: flutter

 collection: ^1.16.0   flutter_dotenv: ^5.0.2   mobx: ^2.0.7+4   flutter_mobx: ^2.0.6+1   routemaster: ^1.0.1   webview_flutter: ^3.0.4   json_annotation: ^4.6.0   provider: ^6.0.3   intl: ^0.17.0   easy_debounce: ^2.0.2+1   html: ^0.15.0   url_launcher: ^6.1.5   hive: ^2.2.3   hive_flutter: ^1.1.0   lottie: ^1.3.0   avatars: ^2.0.2   focus_detector: ^2.0.1   dotted_border: ^2.0.0+2   geocoding: ^2.0.5   geolocator: 8.2.1   device_info_plus: 4.1.0   visibility_detector: ^0.2.2   launchdarkly_flutter_client_sdk: ^1.1.2   new_version: ^0.3.1   cupertino_icons: ^1.0.5   modal_bottom_sheet: ^2.1.0   package_info_plus: ^1.4.2   path: ^1.8.0   chewie: ^1.3.4   video_player: ^2.4.5   youtube_player_flutter: ^8.1.0   cached_network_image: ^3.2.1   animations: ^2.0.3   expandable_page_view: ^1.0.17

dev_dependencies:   flutter_lints: ^2.0.1   json_serializable: ^6.1.1   build_runner: ^2.2.0   flutter_test:     sdk: flutter

flutter:   module:     androidX: true     androidPackage: com.tatadigital.neu_app_embed     iosBundleIdentifier: com.tatadigital.neuAppEmbed   assets:     - assets/logo_512x512.png     - assets/logo_32x32@2x.png

**Affected Devices and CrashCount:**
iPhone12 - 11
iPhone13 - 9
iPhone11 -  6
iPhone 13 Pro -  5
iPhone 12 Pro - 4
iPhone 12 Pro Max- 3
iPhone XS - 2
iPhone 12 Mini -2
iPhone XR - 2
iPhone X - 2
iPhone 7 - 2
iPhone 13 Mini -1
iPhone 13 Pro Max- 1
iPhone SE- 1
iPhone 8 - 1

**Affected OS and CrashCount:**
15.6.1 - 23
16.0.0 - 12
15.6.0 - 7
16.0.2 - 4
15.5.0 - 4
15.5.7 - 2
15.4.1 - 1
14.7.1 - 1

**Comment by huycozy** [comment]
@siddew201 thanks for providing the detail. Labeling it for further investigation from the team.

**Comment by chinmaygarde** [comment]
This is an issue during shell destruction where tearing down the accessibility bridge accesses the view that hasn't been loaded yet. Thereby causing it to load while in the background. The issue is caused by [.view](https://github.com/flutter/engine/blob/109e859595e28fe79b65c09c898849c1acf93733/shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm#L66) access here. It needs to be guarded behind [`-[UIView isViewLoaded]`](https://developer.apple.com/documentation/uikit/uiviewcontroller/2097563-viewloaded?language=objc).

cc @jmagman

**Comment by jmagman** [comment]
[-viewIfLoaded](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621360-viewifloaded?language=objc)

**Comment by chinmaygarde** [comment]
Ooh, thats a new API. I guess this was a mistake a lot of folks made that they added this API in iOS 9.0.

**Comment by siddew201** [comment]
@jmagman what action item required from our side?

**Comment by jmagman** [comment]
None, I have a fix out for review at https://github.com/flutter/engine/pull/36580 and need to track down the test failure. Thanks for the offer, @siddew201!

**Comment by siddew201** [comment]
@jmagman we are getting this crash in new build also. Do we need to flutter upgrade?

**Comment by siddew201** [comment]
@jmagman  Below is the log

Crashed: com.apple.main-thread
0  Flutter                        0x111c4 -[FlutterEngine iosPlatformView] + 53 (weak_ptr.h:53)
1  Flutter                        0x1193c -[FlutterEngine notifyViewControllerDeallocated] + 415 (FlutterEngine.mm:415)
2  Foundation                     0x35594 -[__NSObserver _doit:] + 348
3  CoreFoundation                 0x2a834 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 28
4  CoreFoundation                 0xc6fd4 ___CFXRegistrationPost_block_invoke + 52
5  CoreFoundation                 0x9a1d0 _CFXRegistrationPost + 456
6  CoreFoundation                 0x408ac _CFXNotificationPost + 728
7  Foundation                     0x1b734 -[NSNotificationCenter postNotificationName:object:userInfo:] + 96
8  Flutter                        0x2ee70 -[FlutterViewController deregisterNotifications] + 835 (FlutterViewController.mm:835)
9  Flutter                        0x2eee8 -[FlutterViewController dealloc] + 846 (FlutterViewController.mm:846)
10 CoreFoundation                 0x2987c _CFXNotificationRegistrationDispose + 76
11 CoreFoundation                 0xc3ae4 _CFXNotificationRegistrationBufferDestroy + 56
12 CoreFoundation                 0x408d4 _CFXNotificationPost + 768
13 Foundation                     0x1b734 -[NSNotificationCenter postNotificationName:object:userInfo:] + 96
14 UIKitCore                      0xf5bde4 -[UIApplication _terminateWithStatus:] + 300
15 UIKitCore                      0x6369d4 -[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:] + 128
16 UIKitCore                      0x82c834 -[_UISceneLifecycleMultiplexer forceExitWithTransitionContext:scene:] + 224
17 UIKitCore                      0xf58128 -[UIApplication workspaceShouldExit:withTransitionContext:] + 224
18 FrontBoardServices             0x4efec __63-[FBSWorkspaceScenesClient willTerminateWithTransitionContext:]_block_invoke_2 + 80
19 FrontBoardServices             0xa308 -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 240
20 FrontBoardServices             0x4ef84 __63-[FBSWorkspaceScenesClient willTerminateWithTransitionContext:]_block_invoke + 132
21 libdispatch.dylib              0x3a30 _dispatch_client_callout + 20
22 libdispatch.dylib              0x74e0 _dispatch_block_invoke_direct + 264
23 FrontBoardServices             0xbc70 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 48
24 FrontBoardServices             0xb040 -[FBSSerialQueue _targetQueue_performNextIfPossible] + 220
25 FrontBoardServices             0xf700 -[FBSSerialQueue _performNextFromRunLoopSource] + 28
26 CoreFoundation                 0xbb414 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
27 CoreFoundation                 0xcc1a0 __CFRunLoopDoSource0 + 208
28 CoreFoundation                 0x56f8 __CFRunLoopDoSources0 + 368
29 CoreFoundation                 0xb05c __CFRunLoopRun + 828
30 CoreFoundation                 0x1ebc8 CFRunLoopRunSpecific + 600
31 GraphicsServices               0x1374 GSEventRunModal + 164
32 UIKitCore                      0x514b58 -[UIApplication _run] + 1100
33 UIKitCore                      0x296090 UIApplicationMain + 364
34 TCPApp                         0xd0e634 main + 39 (AppDelegate.swift:39)
35 ???                            0x109069da4 (Missing)

**Comment by jmagman** [comment]
@siddew201 Can I get `flutter doctor -v`?  The fix is in `3.5.0-1.0.pre`.

**Comment by jmagman** [comment]
@siddew201 this is available only in the master channel so far.
